src/data/pokemon/form_change_tables.h

```diff
@@ -800,6 +800,12 @@ static const struct FormChange sUrshifuRapidStrikeFormChangeTable[] =
     {FORM_CHANGE_TERMINATOR},
 };
 
+static const struct FormChange sPalafinZeroFormChangeTable[] =
+{
+    {FORM_CHANGE_BATTLE_SWITCH, SPECIES_PALAFIN_HERO},
```

Comment:
...they should revert at the end of the battle, which it doesn't seem like it's gonna do from looking at the code.

Reply:
It does change back at the end of the batttle